app/models/katello/content_view_puppet_module.rb

```diff
@@ -8,6 +8,7 @@ class ContentViewPuppetModule < Katello::Model
     validates :content_view_id, :presence => true
     validates :name, :uniqueness => { :scope => :content_view_id }, :allow_blank => true
     validates :uuid, :uniqueness => { :scope => :content_view_id }, :allow_blank => true
+    validate :uuid_associated_with_puppet_module
```

Comment:
Ah, that makes sense. This bug is reproduced when a `ContentViewPuppetModule` has a uuid but not a name/author; however, the uuid is not associated with a valid `PuppetModule`.

How do you feel about a validation for  (uuid that is valid with a `PuppetModule`) OR (name and author that is valid with a `PuppetModule`)?